src/python/pants/build_graph/target_filter_subsystem.py

```diff
@@ -0,0 +1,41 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from pants.subsystem.subsystem import Subsystem
+
+
+logger = logging.getLogger(__name__)
+
+
+class TargetFilter(Subsystem):
+  """Filter out targets matching given options.
```

Comment:
Nit - right now "filter out" is correct, but this subsystem seems more general - you could imagine --include-tags and other options growing here. As such the wording perhaps could stand generalization - maybe "Filter targets matching configured criteria.".